scripts/dev/upstream_merge/upstream_merge_and_test.py

```diff
@@ -0,0 +1,268 @@
+            else:
+                fork_url = (
+                    f"https://x-access-token:{token}@github.com/{username}/"
+                    f"{parts[0].split('/')[-1]}.git"
+                )
+            git_obj = GitRepo(
+                local_repo=parts[0],
+                upstream_repo_url=parts[1],
+                upstream_branch=parts[2],
+                local_base_branch=parts[3],
+                upstream_repo_name=upstream_repo_name,
+                fork_name=fork_name,
+                fork_url=fork_url
+            )
+            print(f"\n{git_obj.local_repo.split('/')[-1]}\n")
+            if skip_merge == "True" or skip_merge == "true":
+                print(f"\tUpstream Merge has Been Skipped")
+                merge_report[git_obj] = (0, "Upstream Merge has Been Skipped")
+            else:
+                os.chdir(git_obj.local_repo)                
+                merge_report[git_obj] = merge_upstream(
+                    git_obj, force_checkout, merge_branch_name
+                )
+                os.chdir(current_directory)
+
+    return merge_report
+
+def build_and_test(clean_build, vm_name, snapshot_name):
+    """
+    Build images and run tests on a VM if there are no merge errors.
+
+    :param clean_build: Boolean indicating whether to perform a clean build.
+    :param vm_name: Name of the virtual machine to use for testing.
+    :param snapshot_name: Name of the VM snapshot to restore before testing.
+    :return: A tuple (status_code, message).
+            Returns (0, None) on success, or error details on failure.
+    """
+    success = setup_env_and_build_packages("--org", clean_build)
+    if success[0] != 0:
+        return success
+    success = install_and_test_image(vm_name, snapshot_name)
+    return success
+
+def push_submodules_and_create_PRs(merge_report, merge_branch_name,
+                                    pr_title, pr_description, username):
+    """
+    For each sub-module that was successfully merged and built, push the merge branch
+    to the user's fork and create a pull request targeting the base branch.
+    Updates the merge_report dictionary with the results of the push and PR steps.
+
+    :param build_and_test_details: Tuple indicating the result of build and test steps.
+    :param merge_report: Dictionary mapping GitRepo objects to (status, message) tuples.
+    :param merge_branch_name: The name of the branch to push and create PRs from.
+    :param work_item_id: The work item ID to reference in the PR description.
+    :param username: The GitHub username owning the downstream fork.
+    :return: Updated merge_report dictionary with push and PR results.
+    """
+    push_and_pr_results = {}
+    for git_obj, (status, message) in list(merge_report.items()):
+        if status == 0 and message is not None:
+            current_directory = os.getcwd()
+            os.chdir(git_obj.local_repo)
```

Comment:
Aren't you storing the absolute path of the local_repo in git_obj.local_repo? Why change back to current directory every iteration then?